docs/en/03_Upgrading/index.md

```diff
@@ -37,123 +89,1015 @@ and also checking the `composer.json` files in [recipe-core](https://github.com/
+
+Requiring `conf/ConfigureFromEnv.php` is is no longer necessary. You should remove any references to it in `_config.php`.
+
+The removal of the `_ss_environment.php` file means that conditional logic is no longer available in the environment
+variable set-up process. This encouraged bad practice and should be avoided. If you still require conditional logic early in the bootstrap, this is best placed in the `_config.php` files.
+
+To access environment variables, use the `SilverStripe\Core\Environment::getEnv()` method. To define environment variables, use the `SilverStripe\Core\Environment::setEnv()` method.
+
+### Finalising your environment upgrade
+It's inadvisable to track your `.env` file in your source control system as it might contain sensitive information.
+
+You should ignore the `.env` file by adding an entry to your `.gitignore` file. You can create a sample environment configuration by duplicating your `.env` file as `.env.sample`, and removing sensitive information from it.
+
+You can safely delete your legacy `_ss_environement.php` if you want.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.  
+
+## Step 3 - Namespacing your project (optional) {#step3}
+
+Namespacing your code is an optional step. It is recommended and will help future-proof your code base.
+
+To learn more about PHP namespace:
+* Read the [official Namespace PHP documentation](http://php.net/manual/en/language.namespaces.php)
+* Read the [PSR-4: Autoloader standard](https://www.php-fig.org/psr/psr-4/).
+
+### Before you start namespacing your codebase
+
+You need to choose a root namespace for your project. We recommend following the `Vendor\Package` pattern.
+
+The `Page` and `PageController` classes *must* be defined in the global namespace (or without a namespace).
+
+If you want your codebase to comply with the PSR-4 standard, make sure sub-directories of of your code folder are using the _upper case camel_ naming convention. For example, `mysite/code/page_types` should be renamed to `mysite/code/PageTypes`.
+
+### Automatically namespacing your codebase with the upgrader
 
-The `reorganise` command can automatically rename the `mysite` and `code` folder for you. It will search your code and find any occurence of `mysite`. It won't replace those occurence with `app` however.
+The `add-namespace` command of the [upgrader tool](https://github.com/silverstripe/silverstripe-upgrader/) provides a feature
+to namespace your codebase and to automatic update references to those classes.
 
-After installing, run the upgrader reorganise command:
 ```
-cd ~/my-project-root
-~/.composer/vendor/bin/upgrade-code reorganise --write
+composer global require silverstripe/upgrader
+/var/www/SS_acme > upgrade-code add-namespace "Acme\\Web" ./mysite/code --recursive --write
```

Comment:
Again, `MyProject` rather than `Acme\\Web`

Reply:
The upgrader doesn't like single NS space. Not sure if it's by design or if it's a bug.

I've switch to `App\Web` for now. And I'll discuss with @tractorcow tomorrow.